master/internal/config/config.go

```diff
@@ -191,3 +191,106 @@ type InternalConfig struct {
 	ExternalSessions  model.ExternalSessions `json:"external_sessions"`
 	PrometheusEnabled bool                   `json:"prometheus_enabled"`
 }
+
+func getSchedulerConfig(conf *Config, rpName string) (schdConf *resourcemanagers.SchedulerConfig) {
+	// if not found, fall back to the resource pools config
+	for _, rpConfig := range conf.ResourcePools {
+		if rpConfig.PoolName != rpName {
+			continue
+		}
+		schdConf = rpConfig.Scheduler
+	}
+
+	if schdConf == nil {
+		return schdConf
+	}
+
+	// if not found, fall back to resource manager config
+	return conf.ResourceManager.AgentRM.Scheduler
+	// TODO k8 RM?
+}
+
+func readPreemptionFromScheduler(conf *resourcemanagers.SchedulerConfig) *bool {
+	if conf.Priority == nil {
+		return nil
+	}
+	return &conf.Priority.Preemption
+}
+
+func readPriorityFromScheduler(conf *resourcemanagers.SchedulerConfig) *int {
+	if conf.Priority == nil {
+		return nil
+	}
+	return conf.Priority.DefaultPriority
+}
+
+// ReadPreemptionStatus resolves the desired preemption status for a job.
+func ReadPreemptionStatus(config *Config, rpName string, jobConf interface{}) bool {
+	var jobPreemptible bool
+	switch jobConf.(type) {
+	case *expconf.ExperimentConfig:
+		jobPreemptible = true
+	case *model.CommandConfig:
+		jobPreemptible = false
+	}
+
+	RMPremption := false // wheter the RM supports preemption
+
+	schdConf := getSchedulerConfig(config, rpName)
+	if preemption := readPreemptionFromScheduler(schdConf); preemption != nil {
+		RMPremption = *preemption
+	}
+
+	// TODO k8 RM?
```

Comment:
@eecsliu would these need to be read out of help configuration?

Reply:
You should be able to get these from the `config.ResourceManager.KubernetesRM`. The scheduler is a string, where "preemption" indicates the preemption scheduler. There is no weight, and priority would be 50 by default. K8s doesn't support setting a default priority right now.

Reply:
If you want to read whether or not a job is currently being preempted, then we'd need to go to the k8s api. But for this stuff, you can infer from the config options very easily.